### PR TITLE
Publish cl_img_generate_mipmap extension specification.

### PR DESCRIPTION
**Pull request by Jeremy-Kemp**

### CODE REVIEW

**Review by bashbaug — APPROVED** [review]
LGTM - this one needs a (hopefully minor) rebase before merging though.

File: extensions/cl_img_generate_mipmap.asciidoc [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+    cl_mem src_image,
+    cl_mem dst_image,
+    cl_mipmap_filter_mode_img mipmap_filter_mode,
+    const size_t *array_region,
+    const size_t *mip_region,
+    cl_uint num_events_in_wait_list,
+    const cl_event *event_wait_list,
+    cl_event *event);
+----
+is used to generate mipmap levels from a source image to a destination image. This allows for seamless mipmap level generation without involving the host to generate them, as that would involve reading back the image, generate its mipmap levels and then upload it to the device again. This reduces bandwidth cost of generating mipmap levels as no host copy is needed and may improve the speed of generating	mipmap levels. The source image and destination image can be the same image in order to reduce programming and memory overhead. As with the extension for mipmaps, the size of the image must be a power of 2.
```
Comment (alycm) [inline extensions/cl_img_generate_mipmap.asciidoc:112]:
```suggestion
is used to generate mipmap levels from a source image to a destination image. This allows for seamless mipmap level generation without involving the host to generate them, as that would involve reading back the image, generate its mipmap levels and then upload it to the device again. This reduces bandwidth cost of generating mipmap levels as no host copy is needed and may improve the speed of generating mipmap levels. The source image and destination image can be the same image in order to reduce programming and memory overhead. As with the extension for mipmaps, the size of the image must be a power of 2.
```

Comment (alycm) [reply extensions/cl_img_generate_mipmap.asciidoc:112]: Extremely trivial whitespace thing if you'll be rebasing anyway,

File: extensions/cl_img_generate_mipmap.asciidoc [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+* `CL_INVALID_CONTEXT` if the context associated with _command_queue_, _src_image_ and _dst_image_ are not the same or if the context associated with _command_queue_ and events in event_wait_list are not the same.
+* `CL_INVALID_MIP_LEVEL` if _mip_region[0]_ is greater than _mip_region[1]_, or if _mip_region[0]_ is less than 0, or if _mip_region[1]_ is less than 1, or if _mip_region[1]_ is greater than the number of mipmap levels in _dst_image_.
+* `CL_INVALID_VALUE` if the images are not of the type image array and _array_region_ is not `NULL`, or the images are of type image array and _array_region_ is `NULL`.
+* `CL_INVALID_VALUE` if the images are of type image array and _array_region_ refer to a range outside of the available range in _dst_image_ and _src_image_, or if _array_region[0]_ is greater than _array_region[1]_.
+* `CL_INVALID_EVENT_WAIT_LIST` if _event_wait_list_ is `NULL` and _num_events_in_wait_list_ _ 0, or _event_wait_list_ is not `NULL` and _num_events_in_wait_list_ is 0, or if event objects in _event_wait_list_ are not valid events.
```
Comment (alycm) [inline extensions/cl_img_generate_mipmap.asciidoc:139]:
```suggestion
* `CL_INVALID_EVENT_WAIT_LIST` if _event_wait_list_ is `NULL` and _num_events_in_wait_list_ > 0, or _event_wait_list_ is not `NULL` and _num_events_in_wait_list_ is 0, or if event objects in _event_wait_list_ are not valid events.
```

File: extensions/cl_img_generate_mipmap.asciidoc [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+* `CL_OUT_OF_RESOURCES` if there is a failure to allocate resources required by the	OpenCL implementation on the device.
```
Comment (alycm) [inline extensions/cl_img_generate_mipmap.asciidoc:140]:
```suggestion
* `CL_OUT_OF_RESOURCES` if there is a failure to allocate resources required by the OpenCL implementation on the device.
```

**Review by alycm — APPROVED** [review]
LGTM.

**Comment by Jeremy-Kemp** [comment]
Thanks both. I've made the suggestions and performed the rebase(s).

**Pull request merged**